my-account/app/com/commercetools/sunrise/myaccount/wishlist/SunriseWishlistController.java

```diff
@@ -0,0 +1,86 @@
+
+import com.commercetools.sunrise.framework.controllers.SunriseContentController;
+import com.commercetools.sunrise.framework.controllers.WithQueryFlow;
+import com.commercetools.sunrise.framework.hooks.EnableHooks;
+import com.commercetools.sunrise.framework.reverserouters.SunriseRoute;
+import com.commercetools.sunrise.framework.reverserouters.myaccount.wishlist.WishlistReverseRouter;
+import com.commercetools.sunrise.framework.template.engine.ContentRenderer;
+import com.commercetools.sunrise.framework.viewmodels.content.PageContent;
+import com.commercetools.sunrise.myaccount.MyAccountController;
+import com.commercetools.sunrise.myaccount.wishlist.pagination.WishlistPaginationSettings;
+import com.commercetools.sunrise.myaccount.wishlist.pagination.WishlistProductsPerPageFormSettings;
+import com.commercetools.sunrise.myaccount.wishlist.viewmodels.WishlistPageContentFactory;
+import com.commercetools.sunrise.search.pagination.EntriesPerPageFormSettings;
+import com.commercetools.sunrise.search.pagination.PaginationSettings;
+import io.sphere.sdk.shoppinglists.ShoppingList;
+import play.libs.concurrent.HttpExecution;
+import play.mvc.Http;
+import play.mvc.Result;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * This controller is used to view the current wishlist.
+ */
+public class SunriseWishlistController extends SunriseContentController implements MyAccountController, WithQueryFlow<Wishlist> {
+    private final WishlistFinderBySession wishlistFinder;
+    private final ClearWishlistControllerAction controllerAction;
+    private final WishlistPageContentFactory wishlistPageContentFactory;
+
+    private final PaginationSettings paginationSettings;
+    private final EntriesPerPageFormSettings entriesPerPageFormSettings;
+
+    @Inject
+    protected SunriseWishlistController(final WishlistFinderBySession wishlistFinder,
+                                        final ClearWishlistControllerAction controllerAction,
+                                        final ContentRenderer contentRenderer,
+                                        final WishlistPageContentFactory wishlistPageContentFactory,
+                                        final WishlistPaginationSettings paginationSettings,
+                                        final WishlistProductsPerPageFormSettings entriesPerPageFormSettings) {
+        super(contentRenderer);
+        this.wishlistFinder = wishlistFinder;
+        this.controllerAction = controllerAction;
+        this.wishlistPageContentFactory = wishlistPageContentFactory;
+        this.paginationSettings = paginationSettings;
+        this.entriesPerPageFormSettings = entriesPerPageFormSettings;
+    }
+
+    @EnableHooks
+    @SunriseRoute(WishlistReverseRouter.WISHLIST_PAGE_CALL)
+    public CompletionStage<Result> show(final String languageTag) {
+        return wishlistFinder.getOrCreate()
```

Comment:
Even though this solution is more direct, it has a big drawback compared to having the finder only finding and the creator only creating: you are creating a shopping list every time somebody accesses his/hers. And it happens the same in the remove line item or clear actions.
As with carts, we have to be careful not to create unnecessary resources. So only when an item is added to the wishlist, this should be created if it doesn't yet exist.
Why not use the same system as used in the rest of the controllers of having the interface `WithRequiredWishlist` (e.g. https://github.com/commercetools/commercetools-sunrise-java/blob/master/shopping-cart/app/com/commercetools/sunrise/framework/WithRequiredCart.java), which can be reused for other situations and forces to define a method to handle not found situations. In the case of the add line item, this situation will call the `WishlistCreator` and try the `process` method again.
Before we had it like you just did, but then devs had to copy paste the contents of this method and couldn't decide what to do in case of not finding it. Now they know they can always control it under the `handleNotFoundX` :)
But let me know if you see problems about it that I fail to see or if you have better suggestions.